geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java

```diff
@@ -774,6 +775,14 @@ private void setServerNames(SSLParameters modifiedParams, HostAndPort addr) {
     List<SNIServerName> oldNames = modifiedParams.getServerNames();
     oldNames = oldNames == null ? Collections.emptyList() : oldNames;
     final List<SNIServerName> serverNames = new ArrayList<>(oldNames);
+
+    if (serverNames.stream()
+        .mapToInt(SNIServerName::getType)
+        .anyMatch(type -> type == StandardConstants.SNI_HOST_NAME)) {
+      // we already have a SNI hostname set. Do nothing.
+      return;
+    }
+
```

Comment:
Two things:

1. a nit: the allocation can be moved after the conditional :)
2. is the type check sufficient or is a content check necessary?

on (2) I wonder if there is a scenario where SNI might be set already, but maybe it _disagrees_ with `addr.getHostName()`

Reply:
For (2) - it certainly could disagree since the JDK is setting it to whatever it wants ... But the JDK also only allows one SNI name of a given type. So I guess we just need to decide what the behavior is that we want. I thought it would best to leave the JDK setting in place.

Reply:
Maybe we could unconditionally delete whatever is there and replace it with our own content?

Reply:
Having looked at the path into this method, it's clear that this framework owns this socket and the only way it can set set will cause it to have the value we expect so I'm ok w/ this code as-is. Thanks for playing!